test/evergreen/configure_combinations.sh

```diff
@@ -64,8 +66,22 @@ for flag in $flags ; do
                        echo "*** ERROR: $flag, $option"
                fi
                IFS="$cr_IFS"
-               echo hello
        done
 done
 IFS=$saved_IFS
-exit $ecode
+if [ $ecode -ne 0 ]; then
+        exit $ecode
+fi
+
+# Handle special build combination for running all the diagnostic tests.
+flag=`CFLAGS=\"-g -Werror\"`
+option=`--enable-silent-rules --enable-diagnostic --disable-static`
+echo "Building with: $flag $option"
+cd "$curdir"
+rm -rf ./build
+mkdir build
+cd ./build
+../configure $flag $option || { echo "*** ERROR: $flag, $option " ; exit 1; }
+make all "$@" || { echo "*** ERROR: $flag, $option " ; exit 1; }
+make check VERBOSE=1 || { echo "*** ERROR: $flag, $option " ; exit 1; }
+exit 0
```

Comment:
As this part is special in running full make check, and there's functions we can reuse to simplify the code, I'd prefer to move this part outside - into 2 separate tasks - say `compile-warnings-to-errors` and `make-check-warnings-to-errors`.